ui/pages/login.tsx

```diff
@@ -96,27 +96,14 @@ const Login: NextPage = () => {
 
           return Promise.reject(err);
         }),
-    [flow, router]
+    [flow, router],
   );
+  const reqDomain = flow?.return_to ? new URL(flow.return_to).hostname : "";
```

Comment:
This is not correct. This `return_to` always points to Hydra, to get info about the client you would have to look into `flow.oauth2_login_request.client`. There are 2 fields there that we can use `client_name` which is supposed to be the human readable name of the client and `client_uri` which is the client's homepage (see https://openid.net/specs/openid-connect-registration-1_0.html#ClientMetadata for more info)

Reply:
Good point, thanks for the clarification. I changed this to construct a string with client name and client url if present.